bciers/apps/compliance/src/app/widgets/BccrHoldingAcountWidget.tsx

```diff
@@ -0,0 +1,96 @@
+import { WidgetProps } from "@rjsf/utils";
+import { useState, useCallback } from "react";
+import { TextField, InputAdornment, CircularProgress } from "@mui/material";
+import debounce from "lodash/debounce";
+import ErrorCircle from "@bciers/components/icons/ErrorCircle";
+import { actionHandler } from "@bciers/actions";
+import CheckCircleIcon from "@mui/icons-material/CheckCircle";
+import { BccrAccountDetailsResponse } from "@/compliance/src/app/components/compliance-summaries/types";
+
+// Helper functions
+const isValidAccountIdFormat = (accountId: string): boolean =>
+  accountId.length === 15 && /^\d{15}$/.test(accountId);
+
+const fetchAccountDetails = async (accountId: string) =>
+  actionHandler(`compliance/bccr/accounts/${accountId}`, "GET");
+
+const isValidResponse = (response: BccrAccountDetailsResponse): boolean =>
+  !response.error && response.tradingName != null;
+
+const BCCRHoldingAccountWidget = (props: WidgetProps) => {
```

Comment:
What's the different between component and widget?

Reply:
Widgets in the rjsf ecosystem are a way to define custom components: https://rjsf-team.github.io/react-jsonschema-form/docs/usage/widgets/
We typically use existing widgets throughout the app, but when a widget has behavior or structure that's significantly different, we should create a new custom widget.